oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/FulltextIndex.java

```diff
@@ -307,7 +307,7 @@ public static String rewriteQueryText(String textsearch) {
                     escaped = false;
                 }
                 rewritten.append(c);
-            } else if (Chars.contains(QUERY_OPERATORS, c)) {
+            } else if (QUERY_OPERATORS.indexOf(c) >= 0) {
```

Comment:
I would convert `QUERY_OPERATORS` to `Set<Character>` and then use `Set.contains`.

Reply:
That would be more expensive. A `Set<Character>` requires boxing the elements. And a call to `Set.contains` will box the argument and calculate an hash. A linear search on a character array, as String.indexOf does, uses much less logic per call and will certainly be faster for an array of just 6 elements (all contiguous in memory).